lib/pure/strutils.nim

```diff
@@ -2050,97 +2050,145 @@ func contains*(s: string, chars: set[char]): bool =
   ## * `find func<#find,string,set[char],Natural,int>`_
   return find(s, chars) >= 0
 
-func replace*(s, sub: string, by = ""): string {.rtl,
+func replace*(s: string, sub, by: char, max = -1): string {.rtl,
+    extern: "nsuReplaceChar".} =
+  ## Replaces `max` occurrences of the character `sub` in `s` with
+  ## the character `by`.
+  ## If `max == -1`, it replaces all occurrences.
+  ##
+  ## Optimized version of `replace <#replace,string,string,string>`_ for
+  ## characters.
+  ##
+  ## See also:
+  ## * `find func<#find,string,char,Natural,int>`_
+  ## * `replaceWord func<#replaceWord,string,string,string>`_
+  ## * `multiReplace func<#multiReplace,string,varargs[]>`_
+  runnableExamples:
+    assert "valid variable name".replace(' ', '_') == "valid_variable_name"
+    assert "Faabar!".replace('a', 'o', 2) == "Foobar!"
+  result = newString(s.len)
+  var occLeft = max
+  for i in 0..s.high:
+    if occLeft != 0 and s[i] == sub:
```

Comment:
this is inefficient; occLeft == 0, we should break

ditto with other procs

Reply:
I think the other ones already did break...
I didn't break here, since we needed to add the rest of the string either way. But now changed it.